### ISSUE TITLE
[Question] What is the purpose of asCommonFlow() in FlowHelpers?

### ISSUE THREAD

**Issue by emreakcan** [kuuuurt/jokes-app-multiplatform#5]
Also, IDE says `watch()` is not used, is it being used in runtime? 
And please can you explain the `CoroutineScope(mainDispatcher + job)` ?

**Comment by kuuuurt** [comment]
Thanks for your question!

> What is the purpose of asCommonFlow() in FlowHelpers?

Currently, there's no way to use `Flow` in iOS which is why we have a workaround ([from KotlinConf](https://github.com/JetBrains/kotlinconf-app/blob/master/common/src/mobileMain/kotlin/org/jetbrains/kotlinconf/FlowUtils.kt)) that exposes a `watch` function which handles the observation and closing of the `Flow` so we can use it in iOS.

> Also, IDE says watch() is not used, is it being used in runtime?

Yes the `watch` function will not be used inside an IntelliJ IDE but it is used in Xcode. Check it out [here](https://github.com/kuuuurt/jokes-app-multiplatform/blob/develop/ios/JokesApp/ViewController.swift#L30)

> And please can you explain the CoroutineScope(mainDispatcher + job)

Basically, this just says to use the main thread when using coroutines. I'd advise you to read more on Coroutines if you do not yet understand this. Read about dispatchers and how we can switch threads with them.

**Comment by emreakcan** [comment]
@kuuuurt  thanks for quick reply ! 
Actually I'm confused about `mainDispatcher + job`  + seems like an extension there?

Creates the CoroutineScope, runs in main thread and runs that job?

--

If we don't care about ios, can we use it like that    ` val jokes: Flow<List<Joke>> get() = _jokes` what would change for android?

**Comment by kuuuurt** [comment]
> Actually I'm confused about mainDispatcher + job + seems like an extension there?

Basically, what it does is it creates a merged context for both the `mainDispatcher` which is `Dispatchers.Main` and `job`. Both are `CoroutineContext` objects so they can be added
Check these out:
https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.coroutines/-coroutine-context/plus.html
https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.coroutines/-coroutine-context/plus.html
https://medium.com/@elizarov/coroutine-context-and-scope-c8b255d59055

> If we don't care about ios, can we use it like that val jokes: Flow<List<Joke>> get() = _jokes what would change for android?

Yes and you don't have to change anything! The workaround is really just for iOS. Everything else works as is. It's still a `Flow` in the JVM/Android world.
